improver/cube_combiner.py

```diff
@@ -277,6 +277,14 @@ def process(
         if self.operation == "mean":
             result.data = result.data / len(cube_list)
 
+        # Check resulting dtype and modify if necessary
+        if result.dtype == np.float64:
+            unique_cube_types = set([c.dtype for c in cube_list])
+            raise TypeError(
+                f"Operation {self.operation} on types {unique_cube_types} results in "
+                "float64 data which cannot be safely coerced to float32"
+            )
+
```

Comment:
Could you add some information to the docstring please: at least to the `Raises` section.  Users expecting this plugin to be generic are going to be confused by this error.

Reply:
Thanks. I'd missed this. Added to doc-string.